dlog/commitment/src/combine.rs

```diff
@@ -64,6 +65,36 @@ fn add_pairs_in_place<P: SWModelParameters>(pairs: &mut Vec<SWJAffine<P>>) {
     }
 }
 
+fn batch_add_assign_no_branch<P: SWModelParameters>(
+    denominators: &mut [P::BaseField],
```

Comment:
It seems scary/error-prone to use `denominators` as a shared buffer like this.
Can you rework this as a private type `FieldBuffer<T>` that has a method `use::<A, F: Fn<&mut T, A>>(mut self: Buffer<T>, f: F) -> A`?  The emitted code should be the same, but it would be more self-explanatory and less prone to errors after sharing.